pkg/commands/artifact/k8s.go

```diff
@@ -0,0 +1,268 @@
+package artifact
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/cheggaaa/pb/v3"
+	"github.com/urfave/cli/v2"
+	"golang.org/x/xerrors"
+	"gopkg.in/yaml.v2"
+
+	"github.com/aquasecurity/fanal/analyzer"
+	"github.com/aquasecurity/fanal/cache"
+	"github.com/aquasecurity/trivy-db/pkg/db"
+	"github.com/aquasecurity/trivy/pkg/log"
+	pkgReport "github.com/aquasecurity/trivy/pkg/report"
+	k8sReport "github.com/aquasecurity/trivy/pkg/report/k8s"
+	"github.com/aquasecurity/trivy/pkg/types"
+
+	"github.com/aquasecurity/trivy-kubernetes/pkg/artifacts"
+	"github.com/aquasecurity/trivy-kubernetes/pkg/k8s"
+	"github.com/aquasecurity/trivy-kubernetes/pkg/trivyk8s"
+)
+
+// K8sRun runs scan on kubernetes cluster
+func K8sRun(ctx *cli.Context) error {
+	opt, err := initOption(ctx)
+	if err != nil {
+		return xerrors.Errorf("option error: %w", err)
+	}
+
+	if err = log.InitLogger(opt.Debug, true); err != nil {
+		return err
+	}
+
+	cacheClient, err := initCache(opt)
+	if err != nil {
+		if errors.Is(err, errSkipScan) {
+			return nil
+		}
+		return xerrors.Errorf("cache error: %w", err)
+	}
+	defer cacheClient.Close()
+
+	// Disable DB update when using client/server
+	if opt.RemoteAddr == "" {
+		if err = initDB(opt); err != nil {
+			if errors.Is(err, errSkipScan) {
+				return nil
+			}
+			return xerrors.Errorf("DB error: %w", err)
+		}
+		defer db.Close()
+	}
+
+	kubeConfig, err := k8s.GetKubeConfig()
+	if err != nil {
+		return xerrors.Errorf("get kubeconfig error: %w", err)
+	}
+
+	k8sDynamicClient, err := k8s.NewDynamicClient(kubeConfig)
+	if err != nil {
+		return xerrors.Errorf("failed to instantiate dynamic client: %w", err)
+	}
+
+	trivyk8s := trivyk8s.New(k8sDynamicClient)
+	if len(opt.KubernetesOption.Namespace) > 0 {
```

Comment:
This condition is redundant as there is a check already in `trivy-kubernetes` :
https://github.com/aquasecurity/trivy-kubernetes/blob/14575687ba8c20fc5281154b944f268e105637ab/pkg/trivyk8s/trivyk8s.go#L54